frontend/app/components/SpotifyPlayer.tsx

```diff
@@ -0,0 +1,803 @@
+  };
+  artists: SpotifyArtist[];
+}
+
+interface QueueState {
+  prevTrack2: SpotifyTrack | null;
+  prevTrack: SpotifyTrack | null;
+  currentTrack: SpotifyTrack | null;
+  nextTrackInQueue: SpotifyTrack | null;
+  nextTrackInQueue2: SpotifyTrack | null;
+  nextTrackInQueue3: SpotifyTrack | null;
+  nextTrackInQueue4: SpotifyTrack | null;
+}
+
+
+export default function SpotifyPlayer() {
+  const [player, setPlayer] = useState<any>(null);
+  const [isPlaying, setIsPlaying] = useState(false);
+  const [currentTrack, setCurrentTrack] = useState<any>(null);
+  const [prevTrack, setPrevTrack] = useState<any>(null);
+  const [prevTrack2, setPrevTrack2] = useState<any>(null);
+  const [nextTrackInQueue, setNextTrackInQueue] = useState<any>(null);
+  const [nextTrackInQueue2, setNextTrackInQueue2] = useState<any>(null);
+  const [nextTrackInQueue3, setNextTrackInQueue3] = useState<any>(null);
+  const [nextTrackInQueue4, setNextTrackInQueue4] = useState<any>(null);
+  const [isAnimating, setIsAnimating] = useState(false);
+  const [animationDirection, setAnimationDirection] = useState<'left' | 'right' | 'up' | 'down' | null>(null);
+  const [volume, setVolume] = useState(50);
+  const [deviceId, setDeviceId] = useState<string | null>(null);
+  const [isLoading, setIsLoading] = useState(true);
+  const [error, setError] = useState<string | null>(null);
+  const [currentFlow, setCurrentFlow] = useState<'flow1' | 'flow2' | 'flow3'>('flow1');
+  const playerRef = useRef<any>(null);
+  const scriptRef = useRef<HTMLScriptElement | null>(null);
+  const [showAlternateFlows, setShowAlternateFlows] = useState(false);
+  const [alternateFlow1, setAlternateFlow1] = useState<QueueState | null>(null);
+  const [alternateFlow3, setAlternateFlow3] = useState<QueueState | null>(null);
+
+  // Utility function to randomly shuffle an array using Fisher-Yates algorithm
+  const shuffleArray = <T,>(array: T[]): T[] => {
+    const newArray = [...array];
+    for (let i = newArray.length - 1; i > 0; i--) {
+      const j = Math.floor(Math.random() * (i + 1));
+      [newArray[i], newArray[j]] = [newArray[j], newArray[i]];
+    }
+    return newArray;
+  };
+
+  // Toggle play/pause state and ensure playback is on the correct device
+  const togglePlay = useCallback(async () => {
+    if (!player || !deviceId) return;
+    
+    const supabase = createClient();
+    const { data: { session } } = await supabase.auth.getSession();
+    
+    if (!session?.provider_token) {
+      setError('No Spotify access token found. Please sign in with Spotify.');
+      return;
+    }
+
+    try {
+      const response = await fetch('https://api.spotify.com/v1/me/player', {
+        headers: {
+          'Authorization': `Bearer ${session.provider_token}`
+        }
+      });
+
+      if (response.status === 204 || 
+          (response.ok && (await response.json()).device?.id !== deviceId)) {
+        await transferPlayback(deviceId, session.provider_token);
```

Comment:
Hm I'm not sure we want to transfer playback to this device - we haven't really defined if we want to be the audio output device or just a controller. Fine for now, but we should decide how we want this experience to work